### PR TITLE
feature/1593 - Add search bar in airport list

### LINKED ISSUE #1593: Suggestion: Search Bar for Airport Picker
Browser: N/A

Airport: All

Approximate play time before issue: N/A

#### Steps to reproduce
- Not an issue

As the list of airports grows, is there any interest in adding a searchbar to the top of the airport list that lets you filter through the available ones?

This would search by ICAO code or airport name.

### PR DESCRIPTION
**Pull request by JPNite3**
Resolves #1593 

The purpose of this pull request is to add a search bar to the top of the airport list for the user to search for airports.

### CODE REVIEW

File: src/assets/scripts/client/ui/UiController.js [hunk 1]
```diff
@@ -662,6 +677,24 @@ class UiController {
         this.$switchAirport.addClass(SELECTORS.CLASSNAMES.ACTIVE);
     }
 
+
+    /**
+     * @for UiController
+     * @method onInitiateAirportSearch
+     */
+
```
Comment (n8rzz) [inline src/assets/scripts/client/ui/UiController.js:685]: remove empty space

File: src/assets/scripts/client/ui/UiController.js [hunk 2]
```diff
@@ -662,6 +677,24 @@ class UiController {
...
+    onInitiateAirportSearch() {
+        EventTracker.recordEvent(TRACKABLE_EVENT.AIRPORTS, 'airport-search', 'start');
+
+        const value = this.$airportDialog.find(SELECTORS.DOM_SELECTORS.AIRPORT_SEARCH).val().toLowerCase();
+
+        $('.dialog-body li').each(
```
Comment (n8rzz) [inline src/assets/scripts/client/ui/UiController.js:691]: This isn't the correct place for this, as you guessed, but you're close.  I'll follow up with some more info on where this should actually live.

**Review by n8rzz — DISMISSED** [review]
I'll see if I can find where that filter should live, but overall looking pretty good.

File: src/assets/scripts/client/ui/UiController.js (hunk 2 above)
Comment (tomomasuda) [reply src/assets/scripts/client/ui/UiController.js:691]: Thank you!

**Comment by j-651** [comment]
@tomomasuda Since there are custom `:focus` styles applied, I'd recommend setting `outline: 0` for the new input.

Also, it might be helpful to show a simple "No airports found" message when the query does not match any of the available airports.

File: src/assets/scripts/client/ui/UiController.js (hunk 2 above)
Comment (j-651) [reply src/assets/scripts/client/ui/UiController.js:690]: @n8rzz do you have a suggestion as to where this should be moved?

**Review by erikquinn — APPROVED** [review]

**Pull request merged**